waku/v2/protocol/waku_lightpush/waku_lightpush.nim

```diff
@@ -98,18 +99,21 @@ proc init*(T: type PushRPC, buffer: seq[byte]): ProtoResult[T] =
   ok(rpc)
 
 # Protocol -------------------------------------------------------
-proc init*(T: type WakuLightPush, peerManager: PeerManager, rng: ref BrHmacDrbgContext, handler: PushRequestHandler): T =
+proc init*(T: type WakuLightPush, peerManager: PeerManager, rng: ref BrHmacDrbgContext, handler: PushRequestHandler, relay: WakuRelay = nil): T =
+  debug "init"
   new result
   result.rng = crypto.newRng()
   result.peerManager = peerManager
   result.requestHandler = handler
+  result.relayReference = relay
```

Comment:
The `relayReference` is interesting here. Wondering if we could do something similar with the other protocols to not have to use the `MessageNotificationSubscription` indirection between protocols.

Reply:
yeah probably, that's what I suggested back when that thing was designed

Reply:
there might be some trade-offs with ordering of mounts etc, and probably want to collect references in a cleaner way